_plugins/show.rb

```diff
@@ -31,60 +52,106 @@ def get_playwright_formatted_long(show)
       return ret
     end
 
-    # Main generation method
-    def generate(site)
-      puts "Processing shows..."
-      all_shows = site.collections["shows"].docs
-      @years = site.collections["years"].docs
-      @people = site.collections["people"].docs
+    def get_show_override_assets(show)
+      show.data["assets"].select { |i| i["display_image"] == true }
+    end
 
-      # This is here as it's needed here, I expect year.rb to be run after.
+    def get_show_asset_type(show, key)
+      show.data["assets"].select { |i|
+        if i.has_key?("page")
+          i["type"] == key and not i["page"] > 1
+        else
+          i["type"] == key
+        end
+      }
+    end
 
-      years_by_slug = generate_years_by_slug(@years)
-      people_by_filename = generate_people_by_filename(@people)
+    def get_show_display_image(show)
+      # Assets required
+      return nil if not show.data.has_key?("assets")
+
+      # Find assets that fit the criteria for being a display image
+      override_assets = get_show_override_assets(show)
+      posters = get_show_asset_type(show, "poster")
+      flyers = get_show_asset_type(show, "flyer")
+      programmes = get_show_asset_type(show, "programme")
+
+      # Pick one in this order of precedence
+      if override_assets.size > 0
+        return override_assets[0]
+      elsif posters.size > 0
+        return posters[0]
+      elsif flyers.size > 0
+        return flyers[0]
+      elsif programmes.size > 0
+        return programmes[0]
+      else
+        return nil
+      end
+    end
 
-      # Compute extra show data attributes
-      shows_by_year = Hash.new
-      for show, index in all_shows.each_with_index do
-        path_split = show.path.split("/")
-        year = path_split[path_split.length-2]
+    def get_show_smugmug(show)
+      if show.data.has_key? "smugmug"
+        smug = Smug.new
+        return smug.get_show_photos(show.data["smugmug"], @site)
+      else
+        return nil
+      end
+    end
+
+    def get_show_legacy_paths(show)
+      "shows/#{show.data["year"]}/#{show.basename_without_ext}.html"
+    end
+
+    def generate_show_pls(show, key)
+      if show.data.has_key?(key) and show.data[key]
+        return parse_person_list(show.data[key], people_by_filename)
+      else
+        return nil
+      end
+    end
 
-        # Set meta attributes
-        show.data["playwright_type"], show.data["playwright"],
-          show.data["playwright_formatted"] = get_playwright(show)
+    def send_people(show, key)
+      """Send a person_list to the reverse index"""
+      if show.data.has_key?(key) and show.data[key].class == Array
```

Comment:
Use `&&` instead of `and`.<br>`Hash#has_key?` is deprecated in favor of `Hash#key?`.